internal/lib/lib.go

```diff
@@ -1710,6 +1710,32 @@ func GetDefaultSecretForRoutes() string {
 	return DefaultRouteCert
 }
 
+func ValidateSvcforClass(key string, svc *corev1.Service) bool {
+	if svc != nil {
+		if svc.Spec.LoadBalancerClass == nil {
+			if isAviDefaultLBController() {
+				return true
+			} else {
+				utils.AviLog.Warnf("key: %s, msg: LoadBalancerClass is not specified for LB service %s and ako.vmware.com/avi-lb is not default loadbalancer controller", key, svc.ObjectMeta.Name)
+				return false
+			}
+		} else {
+			if *svc.Spec.LoadBalancerClass != AviIngressController {
```

Comment:
Code can further be simplified by removing nested 
`else {if{}else{}} `
block by 
`elif{} else{}`

Reply:
Hi Priya, I've written it this way so that the flow is simple to understand, and also to keep it similar in structure to `ValidateIngressForClass`. If its problematic I can change it.